core/core-impl/src/main/java/ai/timefold/solver/core/impl/constructionheuristic/DefaultConstructionHeuristicPhaseFactory.java

```diff
@@ -125,11 +129,26 @@ private Optional<ListVariableDescriptor<?>> findValidListVariableDescriptor(
                     + listVariableDescriptors + ").");
         }
 
+        ListVariableDescriptor<Solution_> listVariableDescriptor = listVariableDescriptors.get(0);
+        failIfBasicAndListVariablesAreCombinedOnSingleEntity(listVariableDescriptor);
         failIfConfigured(phaseConfig.getConstructionHeuristicType(), "constructionHeuristicType");
         failIfConfigured(phaseConfig.getEntityPlacerConfig(), "entityPlacerConfig");
         failIfConfigured(phaseConfig.getMoveSelectorConfigList(), "moveSelectorConfigList");
 
-        return Optional.of(listVariableDescriptors.get(0));
+        return Optional.of(listVariableDescriptor);
+    }
+
+    private static void failIfBasicAndListVariablesAreCombinedOnSingleEntity(ListVariableDescriptor<?> listVariableDescriptor) {
```

Comment:
I'd argue this needs to happen earlier, before the CH, when the solver config is being processed.

Reply:
Sounds good 👍 

While I'm on it, does it make sense to move the multiple list variables check too?

```
        if (listVariableDescriptors.size() > 1) {
            throw new IllegalArgumentException("Construction Heuristic phase does not support multiple list variables ("
                    + listVariableDescriptors + ").");
        }
```

Reply:
It's easier to ask forgiveness than permission, did that.